libsolidity/formal/SMTChecker.cpp

```diff
@@ -333,27 +333,54 @@ void SMTChecker::endVisit(VariableDeclarationStatement const& _varDecl)
 
 void SMTChecker::endVisit(Assignment const& _assignment)
 {
-	if (_assignment.assignmentOperator() != Token::Assign)
+	static map<Token, Token> const compoundToArithmetic{
+		{Token::AssignAdd, Token::Add},
+		{Token::AssignSub, Token::Sub},
+		{Token::AssignMul, Token::Mul},
+		{Token::AssignDiv, Token::Div}
+	};
+	Token op = _assignment.assignmentOperator();
+	if (op != Token::Assign && !compoundToArithmetic.count(op))
 		m_errorReporter.warning(
 			_assignment.location(),
-			"Assertion checker does not yet implement compound assignment."
+			"Assertion checker does not yet implement this assignment operator."
 		);
 	else if (!isSupportedType(_assignment.annotation().type->category()))
 		m_errorReporter.warning(
 			_assignment.location(),
 			"Assertion checker does not yet implement type " + _assignment.annotation().type->toString()
 		);
-	else if (Identifier const* identifier = dynamic_cast<Identifier const*>(&_assignment.leftHandSide()))
+	else if (auto identifier = dynamic_cast<Identifier const*>(&_assignment.leftHandSide()))
 	{
 		VariableDeclaration const& decl = dynamic_cast<VariableDeclaration const&>(*identifier->annotation().referencedDeclaration);
 		solAssert(knownVariable(decl), "");
-		assignment(decl, _assignment.rightHandSide(), _assignment.location());
-		defineExpr(_assignment, expr(_assignment.rightHandSide()));
+		smt::Expression rightHandSide = compoundToArithmetic.count(op) ?
+			arithmeticOperation(
+				compoundToArithmetic.at(op),
+				currentValue(decl),
+				expr(_assignment.rightHandSide()),
+				_assignment.annotation().type,
+				_assignment.location()
+			) :
+			expr(_assignment.rightHandSide())
+		;
+		defineExpr(_assignment, rightHandSide);
+		assignment(decl, _assignment, _assignment.location());
 	}
 	else if (dynamic_cast<IndexAccess const*>(&_assignment.leftHandSide()))
 	{
-		arrayIndexAssignment(_assignment);
-		defineExpr(_assignment, expr(_assignment.rightHandSide()));
+		smt::Expression rightHandSide = compoundToArithmetic.count(op) ?
```

Comment:
Can this code be combined with the one above?

Reply:
I thought about that, the problem is that we need an `if identifier else if indexAccess` to know which parameters to use, then an `if compound` to know which operation to call, and then another `if identifier else if indexAccess` to write correctly...

Reply:
Attempted to make it look nicer, updated.